### ISSUE TITLE
Not getting the screenshots even when granting permissions in api23.

### ISSUE THREAD

**Issue by jonathanrz** [Karumi/Shot#19]
I posted the question in stackoverflow, all the details are there: https://stackoverflow.com/questions/46126100/facebooks-screenshot-tests-failing-with-api-bigger-then-23

**Comment by pedrovgs** [comment]
I'll review your issue next week @jonathanrz. Could you please indicate the version of the plugin being used in your project and the device or emulator configuration you are using?

**Comment by jonathanrz** [comment]
Pedro,

You can check the code in this branch: https://github.com/jonathanrz/myexpenses-android/tree/shot
The commit where I added the plugin is this: https://github.com/jonathanrz/myexpenses-android/commit/61470dfc4855fa3c68319047b330d36e56cbbb8f

I am using the version: 0.1.2
I tested in an emulator with api 24.

One possible problem: I am using Android Studio 3.0. Now Android generates another APK with the test code. Shot you be using the default apk or the test apk to take the screenshot?

**Comment by pedrovgs** [comment]
@jonathanrz the plugin fails trying to save the screenshots in an API >= 23 because the permission has to be granted in the testing APK and not the APK under test. Using the rule named grant permission test rule does not provide this functionality. This is not supported by the official Facebook library and we don't support it for now :( If you or any other friendly contributor would like to help us this could be an awesome PR.

For now, you can use any device <= API 2 😃

**Comment by pedrovgs** [comment]
You should also review your shot configuration. Your project overrides the application id adding a suffix named ``applicationIdSuffix '.debug'`` but your shot configuration uses:

```
shot {
    appId = "br.com.jonathanzanella.myexpenses"
}
```

Remember to update your shot configuration to 


```
shot {
    appId = "br.com.jonathanzanella.myexpenses.debug"
}
```

before running your tests again using an API <= 22 device 😃

**Comment by jonathanrz** [comment]
@pedrovgs thanks for the help, I think that I fixed the problem with the permissions, but I am constantly receiving this error:

`Execution failed for task ':app:executeScreenshotTests'.
> java.io.FileNotFoundException: (...)/app/screenshots/screenshots-default/metadata.xml`

I updated my branch with the code, I think that you will be able to reproduce the error.

I am still running the test against an api 22 x86 emulator with Android Studio 3.0.
